common/util/iterator_adaptors_test.cc

```diff
@@ -85,5 +86,16 @@ TEST(ReversedViewTest, NonEmptySet) {
   EXPECT_THAT(reversed_view(v), ElementsAre(7, 6, 3));
 }
 
+TEST(ConvertToMutableIteratorTest, TemporaryTemporary) {
+  std::vector<int> v{3, 6, 7};
+  const std::vector<int>::const_iterator const_it = v.cbegin() + 1;
+  auto it = ConvertToMutableIterator(const_it, v.begin());
+  EXPECT_TRUE((std::is_same_v<decltype(it), std::vector<int>::iterator>));
```

Comment:
Could this be a `static_assert`, for compile-time guarantee?

Reply:
Done.